src/main/java/ru/skypro/homework/service/impl/AdServiceImpl.java

```diff
@@ -42,44 +50,47 @@ public AdDTO addAd(CreateOrUpdateAdDTO createAdDTO, Authentication authenticatio
         ad.setDescription(createAdDTO.getDescription());
         ad.setAuthor(user);
         ad.setPrice(createAdDTO.getPrice());
+        ad = adRepository.save(ad);
 
-        return AdMapper.toAdDTO(ad);
+        return adMapper.toAdDTO(ad);
     }
 
     @Override
     public AdsDTO getAllAdsByMe(Authentication authentication) {
         String username = authentication.getName();
         List<Ad> adList = adRepository.getAllByAuthor_Username(username);
-        return AdMapper.toAdsDTO(adList);
+        return adMapper.toAdsDTO(adList);
     }
 
     @Override
     public AdsDTO getAllAds() {
         List<Ad> adList = adRepository.findAll();
-        return AdMapper.toAdsDTO(adList);
+        return adMapper.toAdsDTO(adList);
     }
 
     @Override
     public ExtendedAdDTO getExtendedAdInfo(Integer adId) {
         Ad ad = getAdById(adId);
-        return AdMapper.toExtendedAdDTO(ad);
+        return adMapper.toExtendedAdDTO(ad);
     }
 
     @Override
-    public void removeAd(Integer adId) {
+    @PreAuthorize("hasRole('ROLE_ADMIN') or @securityService.isOwnerAd(authentication, #adId)")
```

Comment:
```suggestion
    @PreAuthorize("hasRole('ADMIN') or @securityService.isOwnerAd(authentication, #adId)")
```
ROLE_ADMIN - это authority и проверять его нужно через метод hasAuthority, а если hasRole, то просто название роли

Reply:
Исправлено на ROLE